app/controllers/comments_controller.rb

```diff
@@ -9,14 +10,23 @@ def create
     @comment = Comment.new(comment_params)
     @comment.user = current_user
     if @comment.save
-      redirect_to post_path(@comment.post_id)
+      if params[:comment_id]
```

Comment:
What's this case handling? Rephrased: whats the domain scenario where a `comment_id` would be present (and conversely, when would it be absent).

Reply:
If the comment belongs to another comment, as far as I could tell I couldn't redirect to its parent post (the ideal behavior) without some convoluted looping up through commentable ids. So here I just redirected to root_path until I find a better solution.

I thought about adding a post_id column to every comment so I can redirect to that, but I wasn't sure if that was good style when using polymorphism here.

Reply:
So, the scenarios are:
1. You comment on a post and get redirected back to the post path
2. You comment on a comment and get redirected back to the root path

Since comments have a show action, would redirecting back to the comment path (the thing you just commented on) work as well?

Reply:
The show for comments just shows that comment, no children comments. It would look like the comment never went through and would be confusing. I'll need a big structural change to improve scenario 2.

Reply:
FWIW, HN's comment#show behavior displays the child comments (i.e https://news.ycombinator.com/item?id=20204542). Not sure how much the goal is to replicate the exact behavior or approximate / customize it.

Reply:
Thanks, I'll replicate that soon and redirect to comment#show.